### ISSUE TITLE
Services Cannot be Started If Secure Admin Enabled and Non-Default Admin Credentials Used

### ISSUE THREAD

**Issue by PeterParker** [fatmcgav/fatmcgav-glassfish#62]
Am I correct in saying that if secure admin has been enabled _and_ non-default admin credentials have been used, then `asadmin` _must_ be given a password file when calling (some/most/all of) its commands?  One such example is creating a service, which includes a script to [start the domain](https://github.com/fatmcgav/fatmcgav-glassfish/blob/3013d7f8f40bcaed2a08a46e048f6ba5ff61df05/templates/glassfish-init-domain-el.erb#L20)).

I have forked the repo and [implemented a solution](https://github.com/PeterParker/fatmcgav-glassfish/commit/492251a3a8ad4135e6b7893fb2f04c8b40fc462b) that works for my specific use-case, but I'm not sure what else this issue affects and therefore what else should be changed in the module.  (E.g. it appears that [`PWFILE` will never be populated](https://github.com/fatmcgav/fatmcgav-glassfish/blob/3013d7f8f40bcaed2a08a46e048f6ba5ff61df05/templates/glassfish-init-instance-el.erb#L17)).

Any thoughts you have would be appreciated.

**Comment by fatmcgav** [comment]
@PeterParker As it happens I've been hitting some issues with password files when running a cluster... 

As far as I'm aware, asadmin will attempt to load from a standard location and the current password file gets put in that location. 

However explicitly setting the password file might be a way around the issues I'm seeing... 

Will try and have a play over the next few days and see what I can come up with...

**Comment by fatmcgav** [comment]
Also, it looks like I pass the `passwordfile` path to most of the defined types, such as [create_cluster.pp](https://github.com/fatmcgav/fatmcgav-glassfish/blob/develop/manifests/create_cluster.pp#L70)

**Comment by PeterParker** [comment]
Yes -- I noticed that too.  I _suspect_ whenever asadmin is used and a custom credentials have been used then passwordfile should also be passed in -- but I'm far from a GlassFish expert.

**Comment by PeterParker** [comment]
Also, my proposed fix for my use-case probably breaks functionality for others since it passes in a passwordfile whether one has been specified or not.
